commit 40f358d1e7b47834f0d2418f4444925f093bdff6 (refs/changes/34/934/1)
Author: Richard Bradley <richard.bradley@joyent.com>
Date:   2016-11-22T15:34:07+00:00 (2 years, 11 months ago)
    
    MANATEE-262 manatee-adm hangs at spinner even after it appears to have been successful

diff --git a/lib/adm.js b/lib/adm.js
index 959680c..7be2b66 100644
--- a/lib/adm.js
+++ b/lib/adm.js
@@ -81,6 +81,15 @@ function _createZkClient(_, _cb) {
 }
 
 
+function _createZfsClient(_, _cb) {
+    var zfsConfig = _.config.postgresMgrCfg.zfsClientCfg;
+    zfsConfig.dbUser = _.config.postgresMgrCfg.dbUser;
+    zfsConfig.log = LOG;
+    _.zfsClient = new zfs(zfsConfig);
+    return (_cb());
+}
+
+
 /**
  * @param {object} opts.zkClient The zk config
  */
@@ -1290,6 +1299,7 @@ function stateBackfill(opts, cb) {
 function rebuild(opts, cb) {
     vasync.pipeline({ funcs: [
         _createZkClient,
+        _createZfsClient,
         _getShards,
         function _findShard(_, _cb) {
             _.shard = path.basename(_.config.shardPath);
@@ -1387,8 +1397,12 @@ function rebuild(opts, cb) {
                 return _cb();
             });
         },
+        function _ensureZfsMounted(_, _cb) {
+            console.error('Ensuring ZFS dataset is mounted');
+            _.zfsClient.assertDataset(_cb);
+        },
         function _deleteDataDir(_, _cb) {
-            console.error('removing zfs dataset');
+            console.error('Removing ZFS dataset\'s contents');
             var cmd = 'rm -rf ' + _.config.postgresMgrCfg.dataDir + '/*';
             exec(cmd, _cb);
         },
